cypress/README.md

```diff
@@ -0,0 +1,370 @@
+describe(`Example test`, () => {
+
+})
+```
+
+Put `before` or `beforeEach` hook into `describe` block. It will be executed before every test (means `beforeEach`) or before all the tests (means `before`) in the suite.
+
+As exapmle our hook opens needed localhost before every test in the suite.
+
+```typescript
+describe(`Example test`, () => {
+    beforeEach(() => {
+        cy.visit('https"//localhost:3001')
+    })
+})
+```
+
+And finnaly add you test case into the `describe` block by adding `it` block.
+
+```typescript
+describe(`Example test`, () => {
+    beforeEach(() => {
+        cy.visit('https"//localhost:3001')
+    })
+
+    it(`Should open localhost`, () => {
+        cy.url().should('include', 'localhost:3001')
+    })
+})
+```
+
+You're done! 😃
+
+Note: Please use existing methods from `common` folder. If you need to add new method please add it to the `common` folder.
+
+For example, how it test looks with usage of common methods that have already exist.
+
+```typescript
+describe(`Example test`, () => {
+    beforeEach(() => {
+        basePage.openLocalhost(3001)
+    })
+
+    it(`Should open localhost`, () => {
+        basePage.checkUrlText(3001)
+    })
+})
+```
+
+### Create methods
+
+If you need to create a new method you need to add it to the `common` folder. As an example you can use `base.ts` file.
+
+Note: Add methods to `base.ts` only if this method is used more than one sample.
+
+If it's not:
+
+Create a methods folder in `sample/e2e/` folder and create `methods.ts` file in it. Then add your method to the `methods.ts` file. It will give you ability to use this method only in the sample where you created it.
+
+Note: Don't forget to import it in your test
+
+```typescript
+import {ExampleMethods} from "../methods/methods";
+```
+
+And then use it in your test
+
+```typescript
+describe(`Example test`, () => {
+    beforeEach(() => {
+        basePage.openLocalhost(3001)
+    })
+
+    it(`Should open localhost`, () => {
+        exampleMethods.checkSomething('Thing😎')
+    })
+})
+```
+
+<h2 align="center">Best Practices</h2>
+
+### Add and Use data-e2e locators
+
+For example, we have two or more elements on the page with similar names, tags and classes. But you need to get only one of them 😢
+
+There is you can use data-e2e attribute. It will help you to get the needed element. You need simply add `data-e2e` attribute to sample html or etc. 
+
+```html
+<div class="form-group">
+  <input
+    type="text"
+    class="form-control"
+    formControlName="name"
+    placeholder="Name"
+    #name
+    data-e2e="FORM_FIELD__NAME"
+  />
+  <div *ngIf="angForm.controls['name'].invalid && (angForm.controls['name'].dirty || angForm.controls['name'].touched)" class="alert alert-danger">
+    <div *ngIf="angForm.controls['name'].errors.required">
+      Name is required.
+    </div>
+  </div>
+</div>
+```
+And use it in your test
+
+```typescript
+basePage.fillField({
+    selector: '[data-e2e="FORM_FIELD__NAME"]',
```

Comment:
please add some info that we need to provide same selectors structure for similar fields, like input, form field etc, so we could reuse it

Reply:
Fixed